src/utils/api.js

```diff
@@ -0,0 +1,108 @@
+import { processServerResponse } from "./promise";
+import { API_KEY } from "./constants";
+
+/*const baseUrl =
+  process.env.REACT_APP_API_URL === "production"
+    ? "https://api.reciperoulette.twilightparadox.com"
+    : "http://localhost:3002"; */
+
+const baseUrl = import.meta.env.VITE_APP_API_URL;
+
+const searchRecipes = async (searchTerm, page) => {
+  try {
+    const url = new URL(`${baseUrl}/api/recipes/search`);
+    url.searchParams.append("searchTerm", searchTerm);
+    url.searchParams.append("page", page.toString());
+
+    const response = await fetch(url.toString());
+    const data = await processServerResponse(response);
+    console.log("Data:", data);
+    return data;
+  } catch (error) {
+    console.error("Error fetching recipes:", error);
+    throw error;
+  }
+};
+
+const getPopularRecipes = async () => {
+  try {
+    const url = new URL(
+      `https://api.spoonacular.com/recipes/random?apiKey=${API_KEY}&number=6`
+    );
+    const response = await fetch(url);
+    const data = await processServerResponse(response);
+    return data;
+  } catch (error) {
+    console.error("Error fetching popular recipes:", error);
+    throw error;
+  }
+};
+
+const getRecipeSummary = async (recipeId) => {
+  const url = new URL(`${baseUrl}/api/recipes/${recipeId}/summary`);
+  const response = await fetch(url);
+  console.log(response);
+
+  if (!response.ok) {
+    throw new Error(`HTTP error. Status: ${response.status}`);
+  }
+  console.log(response);
+  return response.json();
```

Comment:
*  Please, use `processServerResponse` everywhere not to duplicate the `ok` checking